lib/Sema/CSSimplify.cpp

```diff
@@ -1584,8 +1584,10 @@ shouldOpenExistentialCallArgument(
   if (param->isVariadic())
     return None;
 
-  // Look through an inout type on the formal type of the parameter.
-  auto formalParamTy = param->getInterfaceType()->getInOutObjectType();
+  // Look through an inout and optional types on the formal type of the
+  // parameter.
+  auto formalParamTy = param->getInterfaceType()->getInOutObjectType()
+      ->lookThroughSingleOptionalType();
```

Comment:
I’m pleasantly surprised opening for `inout` types is supported. Aren’t they invariant?

Reply:
Yes, but that's okay: we're binding to the underlying type of the existential, and the generic function cannot change that type.